File: OneDrive/Desktop/Employee-Management-System/frontend/lib/features/employee/data/models/employee_model.dart
class Employee {
  final String id;
  final String name;
  final String email;
  final String department;
  final String position;
  final double salary;

  Employee({
    required this.id,
    required this.name,
    required this.email,
    required this.department,
    required this.position,
    required this.salary,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json["_id"] ?? "",
      name: json["name"] ?? "",
      email: json["email"] ?? "",
      department: json["department"] ?? "",
      position: json["position"] ?? "",
      salary: (json["salary"] ?? 0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "email": email,
      "department": department,
      "position": position,
      "salary": salary,
    };
  }
}

File: OneDrive/Desktop/Employee-Management-System/backend/src/models/employee.model.js
import mongoose from "mongoose"

const employeeSchema = new mongoose.Schema(
 {
  name: {
   type: String,
   required: true,
   trim: true
  },

  email: {
   type: String,
   required: true,
   unique: true,
   lowercase: true,
   trim: true
  },

  department: {
   type: String,
   required: true,
   trim: true
  },

  position: {
   type: String,
   default: ""
  },

  salary: {
   type: Number,
   default: 0
  },

  createdBy: {
   type: mongoose.Schema.Types.ObjectId,
   ref: "User"
  }
 },
 {
  timestamps: true
 }
)

export default mongoose.model("Employee", employeeSchema)